torch/nn/modules/module.py

```diff
@@ -2103,20 +2103,22 @@ def load_state_dict(self, state_dict: Mapping[str, Any],
 
         .. warning::
             If :attr:`assign` is ``True`` the optimizer must be created after
-            the call to :attr:`load_state_dict`.
+            the call to :attr:`load_state_dict` unless
+            :func:`~torch.__future__.get_swap_module_params_on_conversion` is ``True``.
 
         Args:
             state_dict (dict): a dict containing parameters and
                 persistent buffers.
             strict (bool, optional): whether to strictly enforce that the keys
                 in :attr:`state_dict` match the keys returned by this module's
                 :meth:`~torch.nn.Module.state_dict` function. Default: ``True``
-            assign (bool, optional): whether to assign items in the state
-                dictionary to their corresponding keys in the module instead
-                of copying them inplace into the module's current parameters and buffers.
-                When ``False``, the properties of the tensors in the current
-                module are preserved while when ``True``, the properties of the
-                Tensors in the state dict are preserved.
+            assign (bool, optional): When ``False``, the properties of the tensors
+                in the current module are preserved while when ``True``, the
+                properties of the Tensors in the state dict are preserved. Additionally,
+                the ``requires_grad`` of :class:`~torch.nn.Parameter`s in the
+                ``state_dict`` are preserved, but tensors in the ``state_dict``
+                will adopt the ``requires_grad`` of the corresponding parameter
+                in the module.
```

Comment:
```suggestion
                properties of the Tensors in the state dict are preserved. The only exception is
                the ``requires_grad`` field of :class:`~torch.nn.Parameter`s for which the 
                value from the module is preserved.
```

Maybe simpler phrasing?